src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -111,6 +111,7 @@ private List<SourceRepository> GetEffectiveSourcesCore(ISettings settings)
         {
             // Take the passed in sources
             var packageSources = new HashSet<string>(Sources, StringComparer.Ordinal);
+            var sourceObjects = new Dictionary<string, PackageSource>(StringComparer.Ordinal);
```

Comment:
Is it possible simply to pre-create `PackageSource` instances out of `Sources` in order to eliminate `packageSources` or turn it to a Dictionary. This would help avoiding redundancy.